query/src/api/rpc/flight_service_stream.rs

```diff
@@ -49,7 +49,7 @@ impl Stream for FlightDataStream {
             Some(Ok(block)) => match block.try_into() {
                 Err(error) => Some(Err(Status::from(error))),
                 Ok(record_batch) => {
-                    let (dicts, values) = serialize_batch(&record_batch, &self.options);
+                    let (dicts, values) = serialize_batch(&record_batch, &[], &self.options);
```

Comment:
```suggestion
                    let ipc_fields = arrow2::io::ipc::write::default_ipc_fields(record_batch.schema());
                    let (dicts, values) = serialize_batch(&record_batch, &ipc_fields, &self.options);
```

I have PRed #774 to allow passing `None` to this field to avoid users having to take this default themselves.

Reply:
emmm, thanks @jorgecarleitao , I passed `&record_batch.schema().fields` as you suggested and it doesn't seem to work properly and still reports errors in the cluster test.

```rust
                    let record_batch: RecordBatch = record_batch;
                    let ipc_fields = common_arrow::arrow::io::ipc::write::default_ipc_fields(
                        &record_batch.schema().fields,
                    );
                    let (dicts, values) =
                        serialize_batch(&record_batch, &ipc_fields, &self.options);
```